File: 백준/Silver/2164. 카드2/카드2.cs
using System;
using System.Collections.Generic;

namespace BAEKJOON
{
    internal class Program
    {
        static void Main(string[] args)
        {
            int a = int.Parse(Console.ReadLine());

            Queue<int> queue = new Queue<int>();

            for (int i = 1; i <= a; i++)
            {
                queue.Enqueue(i);
            }

            while (queue.Count > 1)
            {
                queue.Dequeue(); // 맨 앞 요소 제거
                int front = queue.Dequeue(); // 다음 요소 제거
                queue.Enqueue(front); // 제거한 요소를 맨 뒤에 추가
            }
            // 마지막 남은 요소 출력
            Console.WriteLine(queue.Dequeue());
        }
    }
}

File: 백준/Silver/2164. 카드2/카드2.cc
#include <iostream>
#include <queue>
using namespace std;

int main(){
    queue<int> q;

    int n;
    cin >> n;

    for(int i = 1; i <= n; i++){
        q.push(i);
    }

    while(q.size() > 1){
        q.pop();
        int front = q.front();
        q.pop();
        q.push(front);
    }

    cout << q.front();
}